packages/website/src/pages/play.tsx

```diff
@@ -8,7 +8,11 @@ function Play(): JSX.Element {
     <Layout title="Playground" description="Playground" noFooter={true}>
       <BrowserOnly fallback={<Loader />}>
         {(): JSX.Element => {
-          const Playground = lazy(() => import('../components/Playground'));
+          const Playground = lazy(
+            () =>
+              // @ts-expect-error: This does not follow Node resolution
```

Comment:
Sorry, I don't follow. It seems like `Playground.tsx` is set up pretty reasonably with a default export of a function component. I tried removing `function Playgound`'s return type, asserting it as a `React.ComponentType`, etc. Seems like a bug somewhere that this is being called a type error, no?

Reply:
This is because we are using `module: "NodeNext"`. Think about what it happens when you compile it to CJS (it gets compiled to CJS because there's no `type: "module"`):

- The `Playground` file becomes `export.default = function ...`
- Here we are dynamically importing the entire module, which will make `module.exports` available as the `default`

So in the end, if it's indeed compiled to CJS and hypothetically run in Node, you would get `{ default: { default: [function Playground] } }`. The reason why you don't is because here it's a more ESM-like resolution.

Reply:
A thought: should we switch to `type: "module"` for the website then?

Reply:
That wouldn't work, because some Docusaurus-generated files are CJS. Webpack is actually _stricter_ when you have `type: "module"`. See https://github.com/facebook/docusaurus/pull/7379

Reply:
🙃 classic